core/chains/evm/client/evm_client.go

```diff
@@ -11,36 +11,36 @@ import (
 	evmconfig "github.com/smartcontractkit/chainlink/v2/core/chains/evm/config"
 	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/config/chaintype"
 	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/config/toml"
-	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
 )
 
-func NewEvmClient(cfg evmconfig.NodePool, chainCfg commonclient.ChainConfig, clientErrors evmconfig.ClientErrors, lggr logger.Logger, chainID *big.Int, nodes []*toml.Node, chainType chaintype.ChainType) Client {
-	var primaries []commonclient.Node[*big.Int, *evmtypes.Head, RPCClient]
-	var sendonlys []commonclient.SendOnlyNode[*big.Int, RPCClient]
+func NewEvmClient(cfg evmconfig.NodePool, chainCfg commonclient.ChainConfig, clientErrors evmconfig.ClientErrors, lggr logger.Logger, chainID *big.Int, nodes []*toml.Node, chainType chaintype.ChainType) (Client, error) {
+	var primaries []commonclient.Node[*big.Int, *RPCClient]
+	var sendonlys []commonclient.SendOnlyNode[*big.Int, *RPCClient]
 	largePayloadRPCTimeout, defaultRPCTimeout := getRPCTimeouts(chainType)
+
 	for i, node := range nodes {
 		var ws *url.URL
 		if node.WSURL != nil {
 			ws = (*url.URL)(node.WSURL)
 		}
 		if node.SendOnly != nil && *node.SendOnly {
-			rpc := NewRPCClient(lggr, nil, (*url.URL)(node.HTTPURL), *node.Name, i, chainID,
-				commonclient.Secondary, cfg.FinalizedBlockPollInterval(), cfg.NewHeadsPollInterval(), largePayloadRPCTimeout, defaultRPCTimeout, chainType)
+			rpc := NewRPCClient(cfg, lggr, ws, (*url.URL)(node.HTTPURL), *node.Name, i, chainID,
```

Comment:
We should always ignore ws URL for sendonly nodes. 
An empty WS URL for sendonly is considered valid. However, with this change, such a sendonly rpc will always be marked as unreachable.